blogs/allSettled.md

```diff
@@ -0,0 +1,47 @@
+<a href="https://github.com/kriskowal/q">Q</a> has <a href="https://github.com/kriskowal/q/wiki/API-Reference#promiseallsettled">Q.allSettled</a>.  <a href="https://docs.angularjs.org/api/ng/service/$q">$q</a> does not.  I've noticed a lot of AngularJS devs request $q.allSettled.  So here it is!  <a href="http://jsfiddle.net/0sr65m7t/">JS Fiddle</a>
+
+<!--more-->
+<pre>
+angular.module("qImproved", [])
+.config(function ($provide) {
+  $provide.decorator("$q", function ($delegate) {
+    function allSettled(promises) {
+      var deferred = $delegate.defer(),
+          counter = 0,
+          results = angular.isArray(promises) ? [] : {};
```

Comment:
Yea, in basically all my Javascript APIs now I support both array and hash inputs.

Really, in many APIs I accept a parameter as a: function or promise or array or hash because it's so easy to support with $q.whenFn + $q's awesome API.